idem_azurerm/exec/azurerm/web/app_service_plan.py

```diff
@@ -0,0 +1,349 @@
+
+    """
+
+    if "location" not in kwargs:
+        rg_props = await hub.exec.azurerm.resource.group.get(
+            ctx, resource_group, **kwargs
+        )
+
+        if "error" in rg_props:
+            log.error("Unable to determine location from resource group specified.")
+            return {
+                "error": "Unable to determine location from resource group specified."
+            }
+        kwargs["location"] = rg_props["location"]
+
+    result = {}
+
+    webconn = await hub.exec.azurerm.utils.get_client(ctx, "web", **kwargs)
+
+    if not isinstance(sku, dict):
+        sku = {"name": sku}
+
+    try:
+        planmodel = await hub.exec.azurerm.utils.create_object_model(
+            "web",
+            "AppServicePlan",
+            sku=sku,
+            kind=kind,
+            reserved=reserved,
+            tags=tags,
+            **kwargs,
+        )
+    except TypeError as exc:
+        result = {
+            "error": "The object model could not be built. ({0})".format(str(exc))
+        }
+        return result
+
+    try:
+        plan = webconn.app_service_plans.create_or_update(
+            name=name, resource_group_name=resource_group, app_service_plan=planmodel,
+        )
+
+        plan.wait()
+        result = plan.result().as_dict()
+    except CloudError as exc:
+        await hub.exec.azurerm.utils.log_cloud_error("web", str(exc), **kwargs)
+        result = {"error": str(exc)}
+    except (ValidationError, DefaultErrorResponseException) as exc:
+        result = {"error": str(exc)}
+
+    return result
+
+
+async def delete(hub, ctx, name, resource_group, **kwargs):
+    """
+    .. versionadded:: 3.0.0
+
+    Delete an App Service Plan.
+
+    :param name: The name of the App Service Plan.
+
+    :param resource_group: The name of the resource group.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        azurerm.web.app_service_plan.delete test_name test_group
+
+    """
+    result = False
+    webconn = await hub.exec.azurerm.utils.get_client(ctx, "web", **kwargs)
+
+    try:
+        plan = webconn.app_service_plans.delete(
+            name=name, resource_group_name=resource_group,
+        )
+
+        result = True
+    except CloudError as exc:
+        await hub.exec.azurerm.utils.log_cloud_error("web", str(exc), **kwargs)
+        result = {"error": str(exc)}
+
+    return result
+
+
+async def get(hub, ctx, name, resource_group, **kwargs):
+    """
+    .. versionadded:: 3.0.0
+
+    Get an App Service plan.
+
+    :param name: The name of the App Service Plan.
+
+    :param resource_group: The name of the resource group.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        azurerm.web.app_service_plan.get test_name test_group
+
+    """
+    result = {}
+    webconn = await hub.exec.azurerm.utils.get_client(ctx, "web", **kwargs)
+
+    try:
+        plan = webconn.app_service_plans.get(
+            name=name, resource_group_name=resource_group,
+        )
+
+        if plan:
```

Comment:
this seems like a new workflow for `get` (and it seems you're using it in some other modules here as well). what case couldn't be covered by catching an exception?

Reply:
I had done this because at the time I was under the assumption `get` did not return an error upon failure to find a specific resource. However, I now think that this was because I was misunderstanding the error occurring when I used `idem exec` to run it. I'll be looking into this again and I'll let you know.

Reply:
I just looked into this again. If the SDK get operation for Web Apps encounters an issue like a non-existent resource group or web app, then it just returns None instead of throwing a CloudError like every other get module has done. I just changed the module to catch the AttributeError and throw a custom error, so now it matches every other get module.